src/docs/user_guide.md

```diff
@@ -1258,6 +1258,32 @@ Moreover, there is no need for any mapping in the JSON file.
 `Image` features are transformed into a int8 valued tensor of size `n x h x w x c` (where `n` is the size of the dataset and `h x w` is a specific resizing of the image that can be set, and `c` is the number of color channels) and added to HDF5 with a key that reflects the name of column in the dataset.
 The column name is added to the JSON file, with an associated dictionary containing preprocessing information about the sizes of the resizing.
 
+Datasets API
+--------------
+We now have a working implementation of the DataSet API, this API is heavily driven by configuration and the ability to quickly generate datasets on the fly.  As examples we have implemented the OHSUMed and Reuters datasets.  Each DataSet contains one or more mixins, these mixins typically revolve around three pieces of functionality, being able to download either zipped up datasets or csv implementations of datasets, doing some pre-processing for those datasets and finally converting the datasets to Pandas DataFrames.  The goal of the DataSet API is to provide training datasets that can directly be plugged into Ludwig Machine Learning Models.
+
+#### Example Datasets Currently Available
+* OHSUMed
+* Kaggle
+* Reuters
+* MNist
+
+Please refer to [Python Kaggle Client](https://technowhisp.com/kaggle-api-python-documentation/) to see how you can setup your own Kaggle credentials.
+Also see the internals of the [Kaggle Client](https://github.com/ludwig-ai/ludwig/blob/master/ludwig/datasets/titanic/__init__.py) here: 
+
+#### Unit Tests Around the DataSets API
+* The easiest use of the Datasets API would be the unit tests, to see this, look at this [titanic unit test](https://github.com/ludwig-ai/ludwig/tree/master/tests/ludwig/datasets/titanic/test_titanic_workflow.py)
+* Here's another example of a unit test for [mnist](https://github.com/ludwig-ai/ludwig/blob/master/tests/ludwig/datasets/mnist/test_mnist_workflow.py)
+
+#### How to use the DataSet API to create a new DataSet
+* Override the base_dataset class and the set of mixins you need , here's an example:  class MyCoolDataSet(ZipDownloadMixin, IdentityProcessMixin, CSVLoadMixin, BaseDataset)
```

Comment:
I would say something like:

Override `ludwig.datasets.base_dataset.BaseDataset` and implement the following methods:

```
    @abc.abstractmethod
    def download_raw_dataset(self):
        raise NotImplementedError()

    @abc.abstractmethod
    def process_downloaded_dataset(self):
        raise NotImplementedError()

    @abc.abstractmethod
    def load_processed_dataset(self, split):
        raise NotImplementedError()
```

For common steps (e.g., extracting zip files, downloading from Kaggle, etc.) a set of mixins are available for your subclass.  When adding a new mixin, add its configurable properties within a `config.yaml` file in the new dataset package.